src/python-common/ceph/deployment/drive_group.py

```diff
@@ -303,10 +303,10 @@ def validate(self):
             raise DriveGroupValidationError(f"{self.objectstore} is not supported. Must be "
                                             f"one of ('bluestore')")
 
-        if self.block_wal_size is not None and type(self.block_wal_size) != int:
-            raise DriveGroupValidationError('block_wal_size must be of type int')
-        if self.block_db_size is not None and type(self.block_db_size) != int:
-            raise DriveGroupValidationError('block_db_size must be of type int')
+        if self.block_wal_size is not None and type(self.block_wal_size) not in [int, str]:
+            raise DriveGroupValidationError('block_wal_size must be of type int or string')
+        if self.block_db_size is not None and type(self.block_db_size) not in [int, str]:
+            raise DriveGroupValidationError('block_db_size must be of type int or string')
```

Comment:
If this replaces #35591, feel free to close my PR then!

Reply:
I pulled in the test fixes from #35591 too.